### PR TITLE
feat: Support enableNativeCrashHandling for iOS

### PR DESCRIPTION
**Pull request by marandaneto**
## :loudspeaker: Type of change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring

## :scroll: Description
feat: Support enableNativeCrashHandling for iOS

## :bulb: Motivation and Context
Closes https://github.com/getsentry/sentry-react-native/issues/1362

## :pencil: Checklist

- [X] I reviewed submitted code
- [ ] I added tests to verify changes
- [X] All tests passing
- [X] No breaking changes

### CODE REVIEW

File: ios/RNSentry.m [hunk 1]
```diff
@@ -77,6 +77,15 @@ + (BOOL)requiresMainQueueSetup {
         reject(@"SentryReactNative", error.localizedDescription, error);
         return;
     }
+    
+    BOOL enableNativeCrashHandling = (BOOL)[mutableOptions valueForKey:@"enableNativeCrashHandling"];
```
Comment (bruno-garcia) [inline ios/RNSentry.m:81]: Is this going to default to `@NO` if there's no value in the key?

**Review by bruno-garcia — COMMENTED** [review]
Is this something we can test with the e2e tests?

File: ios/RNSentry.m (hunk 1 above)
Comment (marandaneto) [reply ios/RNSentry.m:81]: No, there's always a value, https://github.com/getsentry/sentry-react-native/blob/main/src/js/sdk.tsx#L29

Comment (marandaneto) [reply ios/RNSentry.m]: I've added a safeguard since one could forcefully `enableNativeCrashHandling: null,`

**Review by philipphofmann — APPROVED** [review]
LGTM 🚀

**Comment by marandaneto** [comment]
> Is this something we can test with the e2e tests?

Unless we have different e2e setups, if we remove this integration, the app won't crash, so technically possible.
Since this is only about removing an integration, unit tests would be enough.

**Pull request merged**